### PR TITLE
engine: migrate status update manager to job queue

### PR DESCRIPTION
**Pull request by mastercactapus**
**Description:**
Migrates status updates to use the job queuing system.

Previously, the status manager would run maintenance operations every tick, followed by each subscription in a tight loop (up to 100 per tick) keeping a slice of IDs to omit (which was passed to the DB for each call).

The new approach:
- hourly periodic job to trim old subscriptions
- every 5 seconds a look-for-work periodic job that schedules jobs for individual subscriptions
- each subscription runs in it's own job in the queue, up to 5 concurrently (management tasks take priority)

**Out of Scope:**
- Adding a trigger from the CreateOrUpdate method of alerts to immediately schedule things like status updates
- With the above ^ we could update the "look for work" job to run once per minute instead

**Describe any introduced user-facing changes:**
N/A

**Describe any introduced API changes:**
N/A

### CODE REVIEW

File: engine/statusmgr/cleanup.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+	"github.com/riverqueue/river"
+	"github.com/target/goalert/gadb"
+)
+
+type CleanupArgs struct{}
+
+func (CleanupArgs) Kind() string { return "status-manager-cleanup" }
+
+func (db *DB) cleanup(ctx context.Context, j *river.Job[CleanupArgs]) error {
```
Comment (AlaricWhitney) [inline engine/statusmgr/cleanup.go:15]: add functional comment to explain the purpose of the function

File: app/initriver.go [hunk 2]
```diff
@@ -42,6 +42,40 @@ type noopWorker struct{}
...
 
+type ignoreCancel struct {
+	h slog.Handler
+}
+
+func (i *ignoreCancel) Enabled(ctx context.Context, level slog.Level) bool {
+	return i.h.Enabled(ctx, level)
+}
+
+func (i *ignoreCancel) Handle(ctx context.Context, rec slog.Record) error {
```
Comment (AlaricWhitney) [inline app/initriver.go:53]: add functional comment to explain the purpose of the function

File: engine/statusmgr/lookforwork.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	"github.com/jackc/pgx/v5"
+	"github.com/riverqueue/river"
+	"github.com/target/goalert/gadb"
+)
+
+type LookForWorkArgs struct{}
+
+func (LookForWorkArgs) Kind() string { return "status-manager-look-for-work" }
+
+func (db *DB) lookForWork(ctx context.Context, j *river.Job[LookForWorkArgs]) error {
```
Comment (AlaricWhitney) [inline engine/statusmgr/lookforwork.go:18]: add functional comment to explain the purpose of the function

File: engine/statusmgr/setup.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+const (
+	QueueName           = "status-manager"
+	PriorityLookForWork = 2
+	PriorityCleanup     = 3
+	PriorityProcess     = 4
+)
+
+var _ processinglock.Setupable = &DB{}
+
+func (db *DB) Setup(ctx context.Context, args processinglock.SetupArgs) error {
```
Comment (AlaricWhitney) [inline engine/statusmgr/setup.go:20]: add functional comment to explain the purpose of the function

**Review by AlaricWhitney — CHANGES_REQUESTED** [review]

**Review by AlaricWhitney — APPROVED** [review]

**Review by AlaricWhitney — APPROVED** [review]

**Review by KatieMSB — DISMISSED** [review]

**Review by AlaricWhitney — APPROVED** [review]

**Review by ethan-haynes — APPROVED** [review]

**Pull request merged**